### PR TITLE
Fixed redundant/ambiguous dispatch methods

### PR DESCRIPTION
**Pull request by huwpascoe**
#### Purpose

Step 1 of 4~ towards unified networking protocol that will simplify networking and drastically improve bandwidth. As it stands the server dispatch consists of multiple ambiguous methods with no clear data path.
#### Review

Easy to test, just play the game. Abnormal behavior compared to upstream/bleed is a bug.
Didn't find any problems? Please give a thumbs up; thank you for your time!

### CODE REVIEW

File: OpenRA.Game/Server/Server.cs [hunk 1]
```diff
@@ -387,35 +387,53 @@ void SetOrderLag()
...
+				{
+					SendData(c.Socket, BitConverter.GetBytes(data.Length + 4));
+					SendData(c.Socket, BitConverter.GetBytes(from));
+					SendData(c.Socket, BitConverter.GetBytes(frame));
+					SendData(c.Socket, data);
+				}
+				catch (Exception)
+				{
+					DropClient(c);
+					Log.Write("server", "Dropping client {0} because dispatching orders failed: {1}");
```
Comment (ScottNZ) [inline OpenRA.Game/Server/Server.cs]: This is missing the extra arguments for {0} and {1}.

Comment (huwpascoe) [inline OpenRA.Game/Server/Server.cs]: Thanks! fixed

**Comment by huwpascoe** [comment]
(this was updated btw)

**Comment by huwpascoe** [comment]
rebased

File: OpenRA.Game/Server/Server.cs [hunk 2]
```diff
@@ -387,35 +387,53 @@ void SetOrderLag()
...
+				{
+					SendData(c.Socket, BitConverter.GetBytes(data.Length + 4));
+					SendData(c.Socket, BitConverter.GetBytes(from));
+					SendData(c.Socket, BitConverter.GetBytes(frame));
+					SendData(c.Socket, data);
+				}
+				catch (Exception e)
+				{
+					DropClient(c);
+					Log.Write("server", "Dropping client {0} because dispatching orders failed: {1}", c, e);
```
Comment (RoosterDragon) [inline OpenRA.Game/Server/Server.cs]: What is the first parameter now the connection object? Will that produce a useful string?

**Comment by RoosterDragon** [comment]
I'm not familiar with the server code much, but this change certainly makes it even harder for me to read by stuffing a lot into one function signature.

It feels like one of those nasty Windows APIs with a dozen parameters where the function does very different things if certain things are null and some combinations of parameters are totally invalid and therefore you have to pass a bunch of nulls in and it's icky.

For example, it accepts both a data and order parameter, but these are used in a mutually exclusive manner. That would be much better as two methods so you can't get confused.

Personally, I feel like you lose sense of where the orders are going by forcing everything into a signle method signature, but that's just me.

**Comment by huwpascoe** [comment]
@RoosterDragon I haven't so much improved it here as revealed what's really going on....This WINAPI style method is the distilled representation of the current state of the networking code, the server messaging part at least. It'll be replaced entirely a few PRs later.

Believe me I'd post the whole thing as one PR if I could, but people just don't seem to want to review networking code. So I have to make it clear as possible in tiny steps, even if they don't make a lot of sense at the time.

**Comment by huwpascoe** [comment]
Actually, nah this way isn't working any better either. I might as well just submit one big PR sometime.

**Pull request closed without merging**